app/code/Magento/CheckoutAgreements/Model/Checkout/Plugin/GuestValidation.php

```diff
@@ -117,7 +158,21 @@ private function isAgreementEnabled()
             AgreementsProvider::PATH_ENABLED,
             ScopeInterface::SCOPE_STORE
         );
-        $agreementsList = $isAgreementsEnabled ? $this->checkoutAgreementsRepository->getList() : [];
+        $storeFilter = $this->filterBuilder
```

Comment:
This piece of code occurs multiple times in this PR along with 3 dependencies to support it. 
It would make sense to extract it to a separate service.

Reply:
It's just workflow with searchCriteria mechanism, I think it will be useless create new service only for couple of filters

Reply:
Introducing this many boilerplate code creates a challenge in maintenance of it.
It is much easier to fix/update only 1 service instead of searching all over the module in the future.